internal/services/eventgrid/eventgrid_namespace_resource_test.go

```diff
@@ -0,0 +1,529 @@
+}
+
+
+resource "azurerm_eventgrid_namespace" "test" {
+  name                = "acctest-egn-%[1]d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+
+  capacity              = 2
+  public_network_access = "Disabled"
+  sku                   = "Standard"
+
+  topic_spaces_configuration {
+    alternative_authentication_name_source          = ["ClientCertificateEmail", "ClientCertificateSubject"]
+    maximum_client_sessions_per_authentication_name = 2
+    maximum_session_expiry_in_hours                 = 2
+    route_topic_id                                  = azurerm_eventgrid_topic.test.id
+
+    dynamic_routing_enrichment {
+      key   = "hello"
+      value = "$${client.authenticationName}"
+    }
+
+    static_routing_enrichment {
+      key   = "hello2"
+      value = "world"
+    }
+  }
+
+  identity {
+    type = "UserAssigned"
+    identity_ids = [
+      azurerm_user_assigned_identity.test.id
+    ]
+  }
+}
+`, data.RandomInteger, data.Locations.Primary)
+}
+
+func (r EventGridNamespaceResource) inboundIpRules(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%[1]d"
+  location = "%[2]s"
+}
+
+resource "azurerm_eventgrid_topic" "test" {
+  name                = "acctesteg-%[1]d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+}
+
+resource "azurerm_eventgrid_namespace" "test" {
+  name                = "acctest-egn-%[1]d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+
+  public_network_access = "Enabled"
+
+  inbound_ip_rule {
+    ip_mask = "10.0.0.0/16"
+    action  = "Allow"
+  }
+
+  inbound_ip_rule {
+    ip_mask = "10.1.0.0/16"
+    action  = "Allow"
+  }
+}
+`, data.RandomInteger, data.Locations.Primary)
+}
+
+func (r EventGridNamespaceResource) inboundIpRulesUpdated(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%[1]d"
+  location = "%[2]s"
+}
+
+resource "azurerm_eventgrid_topic" "test" {
+  name                = "acctesteg-%[1]d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+}
+
+resource "azurerm_eventgrid_namespace" "test" {
+  name                = "acctest-egn-%[1]d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+
+  public_network_access = "Enabled"
+
+  inbound_ip_rule {
+    ip_mask = "10.2.0.0/16"
+    action  = "Allow"
+  }
+}
+`, data.RandomInteger, data.Locations.Primary)
+}
+
+func (r EventGridNamespaceResource) basicWithTags(data acceptance.TestData) string {
```

Comment:
Minor but we could just call this 
```suggestion
func (r EventGridNamespaceResource) tags(data acceptance.TestData) string {
```